src/main/java/com/o19s/es/ltr/feature/PrebuiltFeatureSet.java

```diff
@@ -35,34 +37,46 @@ public String name() {
     }
 
     @Override
-    public List<Feature> features() {
-        return Arrays.asList(features);
+    public PrebuiltFeature[] asArray() {
+        return features.toArray(new PrebuiltFeature[features.size()]);
```

Comment:
Copying a list to an array can get expansive. I wonder if there is a reason to have a different internal vs external representation of features (except to protect the internal state as it is mutable).